[Bug]: r/aws_fsx_ontap_file_system has outdated max iops validation

### Terraform Core Version

1.5.6

### AWS Provider Version

5.14.0

### Affected Resource(s)

* aws_fsx_ontap_file_system

### Expected Behavior

A new FSx ONTAP filesystem created with 160,000 user provisioned IOPS

### Actual Behavior

Terraform fails with error

### Relevant Error/Panic Output Snippet

```shell
❯ terraform apply
╷
│ Error: expected disk_iops_configuration.0.iops to be in the range (0 - 80000), got 160000
│ 
│   with aws_fsx_ontap_file_system.this,
│   on main.tf line 17, in resource "aws_fsx_ontap_file_system" "this":
│   17:     iops = 160000
│ 
╵
```


### Terraform Configuration Files

```hcl
resource "aws_fsx_ontap_file_system" "this" {
  deployment_type = "SINGLE_AZ_1"
  preferred_subnet_id = "subnet-00e7b623d8a8b1d5f"
  subnet_ids      = ["subnet-00e7b623d8a8b1d5f", "subnet-09e9085b776585e30"]
  throughput_capacity = 4096

  disk_iops_configuration {
    iops = 160000
    mode = "USER_PROVISIONED"
  }
}
```

### Steps to Reproduce

Deploy the configuration above

### Debug Output

```
❯ terraform apply
2023-08-31T12:22:11.819-0500 [INFO]  Terraform version: 1.5.6
2023-08-31T12:22:11.820-0500 [DEBUG] using github.com/hashicorp/go-tfe v1.26.0
2023-08-31T12:22:11.820-0500 [DEBUG] using github.com/hashicorp/hcl/v2 v2.16.2
2023-08-31T12:22:11.820-0500 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.0
2023-08-31T12:22:11.820-0500 [DEBUG] using github.com/zclconf/go-cty v1.12.2
2023-08-31T12:22:11.820-0500 [INFO]  Go runtime version: go1.21.0
2023-08-31T12:22:11.820-0500 [INFO]  CLI args: []string{"terraform", "apply"}
2023-08-31T12:22:11.820-0500 [DEBUG] Attempting to open CLI config file: /Users/corey.lane/.terraformrc
2023-08-31T12:22:11.820-0500 [INFO]  Loading CLI configuration from /Users/corey.lane/.terraformrc
2023-08-31T12:22:11.820-0500 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2023-08-31T12:22:11.820-0500 [DEBUG] ignoring non-existing provider search directory /Users/corey.lane/.terraform.d/plugins
2023-08-31T12:22:11.820-0500 [DEBUG] ignoring non-existing provider search directory /Users/corey.lane/Library/Application Support/io.terraform/plugins
2023-08-31T12:22:11.820-0500 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2023-08-31T12:22:11.820-0500 [INFO]  CLI command args: []string{"apply"}
2023-08-31T12:22:11.821-0500 [DEBUG] New state was assigned lineage "8ab13b29-6585-b35d-5a0d-2ce2506eac0f"
2023-08-31T12:22:12.000-0500 [DEBUG] checking for provisioner in "."
2023-08-31T12:22:12.001-0500 [DEBUG] checking for provisioner in "/opt/homebrew/bin"
2023-08-31T12:22:12.002-0500 [INFO]  backend/local: starting Apply operation
2023-08-31T12:22:12.005-0500 [DEBUG] created provider logger: level=debug
2023-08-31T12:22:12.005-0500 [INFO]  provider: configuring client automatic mTLS
2023-08-31T12:22:12.010-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5]
2023-08-31T12:22:12.020-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5 pid=19458
2023-08-31T12:22:12.020-0500 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5
2023-08-31T12:22:12.437-0500 [INFO]  provider.terraform-provider-aws_v5.14.0_x5: configuring server automatic mTLS: timestamp=2023-08-31T12:22:12.437-0500
2023-08-31T12:22:12.442-0500 [DEBUG] provider.terraform-provider-aws_v5.14.0_x5: plugin address: address=/var/folders/wl/sjh4bfys3l1b7vdrphlkdb8r0000gn/T/plugin3019740363 network=unix timestamp=2023-08-31T12:22:12.441-0500
2023-08-31T12:22:12.442-0500 [DEBUG] provider: using plugin: version=5
2023-08-31T12:22:12.963-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-31T12:22:12.965-0500 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5 pid=19458
2023-08-31T12:22:12.966-0500 [DEBUG] provider: plugin exited
2023-08-31T12:22:12.966-0500 [DEBUG] Building and walking validate graph
2023-08-31T12:22:12.966-0500 [DEBUG] adding implicit provider configuration provider["registry.terraform.io/hashicorp/aws"], implied first by aws_fsx_ontap_file_system.this
2023-08-31T12:22:12.966-0500 [DEBUG] ProviderTransformer: "aws_fsx_ontap_file_system.this" (*terraform.NodeValidatableResource) needs provider["registry.terraform.io/hashicorp/aws"]
2023-08-31T12:22:12.966-0500 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/hashicorp/aws\"]" references: []
2023-08-31T12:22:12.966-0500 [DEBUG] ReferenceTransformer: "aws_fsx_ontap_file_system.this" references: []
2023-08-31T12:22:12.966-0500 [DEBUG] Starting graph walk: walkValidate
2023-08-31T12:22:12.966-0500 [DEBUG] created provider logger: level=debug
2023-08-31T12:22:12.966-0500 [INFO]  provider: configuring client automatic mTLS
2023-08-31T12:22:12.968-0500 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5 args=[.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5]
2023-08-31T12:22:12.979-0500 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5 pid=19461
2023-08-31T12:22:12.979-0500 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5
2023-08-31T12:22:13.399-0500 [INFO]  provider.terraform-provider-aws_v5.14.0_x5: configuring server automatic mTLS: timestamp=2023-08-31T12:22:13.399-0500
2023-08-31T12:22:13.403-0500 [DEBUG] provider.terraform-provider-aws_v5.14.0_x5: plugin address: address=/var/folders/wl/sjh4bfys3l1b7vdrphlkdb8r0000gn/T/plugin2234352463 network=unix timestamp=2023-08-31T12:22:13.403-0500
2023-08-31T12:22:13.403-0500 [DEBUG] provider: using plugin: version=5
2023-08-31T12:22:13.848-0500 [ERROR] provider.terraform-provider-aws_v5.14.0_x5: Response contains error diagnostic: diagnostic_attribute=AttributeName("disk_iops_configuration").ElementKeyInt(0).AttributeName("iops") diagnostic_severity=ERROR @caller=github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_detail= diagnostic_summary="expected disk_iops_configuration.0.iops to be in the range (0 - 80000), got 160000" tf_provider_addr=registry.terraform.io/hashicorp/aws @module=sdk.proto tf_proto_version=5.3 tf_req_id=7105bdb7-9c7e-9075-27a0-12e7dc431d3e tf_rpc=ValidateResourceTypeConfig tf_resource_type=aws_fsx_ontap_file_system timestamp=2023-08-31T12:22:13.848-0500
2023-08-31T12:22:13.848-0500 [ERROR] vertex "aws_fsx_ontap_file_system.this" error: expected disk_iops_configuration.0.iops to be in the range (0 - 80000), got 160000
╷
│ Error: expected disk_iops_configuration.0.iops to be in the range (0 - 80000), got 160000
│ 
│   with aws_fsx_ontap_file_system.this,
│   on main.tf line 8, in resource "aws_fsx_ontap_file_system" "this":
│    8:     iops = 160000
│ 
╵
2023-08-31T12:22:13.849-0500 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-08-31T12:22:13.853-0500 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/hashicorp/aws/5.14.0/darwin_arm64/terraform-provider-aws_v5.14.0_x5 pid=19461
2023-08-31T12:22:13.853-0500 [DEBUG] provider: plugin exited
```

### Panic Output

_No response_

### Important Factoids

_No response_

### References

- https://aws.amazon.com/about-aws/whats-new/2022/11/amazon-fsx-netapp-ontap-doubles-maximum-throughput-capacity-ssd-iops-file-system/
- https://docs.aws.amazon.com/fsx/latest/ONTAPGuide/document-history.html

### Would you like to implement a fix?

PR #33263

--- a/.changelog/33263.txt
+++ b/.changelog/33263.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_fsx_ontap_file_system: Increase maximum value of `disk_iops_configuration.iops` to `160000`
+```

--- a/internal/service/fsx/ontap_file_system.go
+++ b/internal/service/fsx/ontap_file_system.go
@@ -80,7 +80,7 @@ func ResourceOntapFileSystem() *schema.Resource {
 							Type:         schema.TypeInt,
 							Optional:     true,
 							Computed:     true,
-							ValidateFunc: validation.IntBetween(0, 80000),
+							ValidateFunc: validation.IntBetween(0, 160000),
 						},
 						"mode": {
 							Type:         schema.TypeString,

